src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -136,7 +136,7 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
-                cms.ComputeSignature(cmsSigner);
+                cms.ComputeSignature(cmsSigner, false); // silent is false to ensure PIN prompts appear if CNG/CAPI requires it
```

Comment:
NuGet.exe by default runs in an interactive mode. 
However, there is a [noninteractive switch](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-sign#options) that can be used.

https://github.com/NuGet/NuGet.Client/blob/502505da868f79b1c4fc1704dae6642138adb30c/src/NuGet.Core/NuGet.Commands/SignCommand/CertificateFindOptions.cs#L52 already propagates that information so it just needs to be wired in. 
Taking a quick look some heavier refactoring might be required.

Reply:
@nkolev92 are you asking me to do this bigger refactoring? I agree it should be done, however I'm not able to do that bigger piece. I'm just trying to preserve current behavior.

Reply:
Please correct me if I'm missing something, but this change would only be relevant in the .NET Core signing case, which is still a WIP by @dtivel and @heng-liu. 

In that case, wouldn't this be better tracked with the rest of the .NET Core signing work?

Reply:
Yes, I wasn't sure the right branch to submit this PR to, it seems like it was this one?

Reply:
Oh, I didn't even see the branch, my bad. 

I'll defer to @heng-liu as the branch owner here. 

It might take some time to receive feedback as she's making her way through all the issues we see :)